utils/logalloc.cc

```diff
@@ -1958,31 +1946,66 @@ const region_impl& region::get_impl() const {
 region::region(region&& other) {
     this->_impl = std::move(other._impl);
     get_impl()._region = this;
+    if (_impl) {
+        auto r_impl = static_cast<region_impl*>(_impl.get());
+        if (r_impl->_listener) {
+            r_impl->_listener->moved(&other, this);
+        }
+    }
 }
 
 region& region::operator=(region&& other) {
+    if (this == &other || _impl == other._impl) {
```

Comment:
Yes - immediately after operator=, this is the case (although we will never reassign again).

I really dislike those subtleties (the thing is done so that lsa compactions notify the memtable reader after we merge to the memtable to cache) and would prefer to remove the assignment, but it is very tricky.